### PR TITLE
DUOS-2778 - Update recipient for 'data-custodian-approval'

### PR DESCRIPTION
**Pull request by hams7504**
### Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2778

### Summary

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/models/Dataset.java [hunk 1]
```diff
@@ -471,4 +477,37 @@ public User getCreateUser() {
...
+            return stringList;
+          })
+          .flatMap(List::stream)
+          .distinct()
+          .toList();
+    }
+    return List.of();
+  }
+
+  public List<String> getDataDepositorEmails() {
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/models/Dataset.java]: Technically, this is getting the string value of the `Data Depositor` field, which could be an email, or not. I suggest renaming it so it isn't confusing to the callers. Related to this, I think the email service handles the case where the string is NOT a valid email, but maybe an extra check in `VoteService` would prevent sending invalid data in the first place.

File: src/main/java/org/broadinstitute/consent/http/service/VoteService.java [hunk 2]
```diff
@@ -362,22 +365,36 @@ protected void notifyCustodiansOfApprovedDatasets(List<Dataset> datasets, User r
...
         });
       }
+
+      // Data Depositors
+      List<String> depositors = d.getDataDepositors();
+      List<String> depositorEmails = new ArrayList<>();
+      String emailRegex = "^[a-zA-Z0-9_+&*-]+(?:\\."+
+          "[a-zA-Z0-9_+&*-]+)*@" +
+          "(?:[a-zA-Z0-9-]+\\.)+[a-z" +
+          "A-Z]{2,7}$";
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/VoteService.java]: I am usually hesitant to implement our own regexes for this kind of thing. It's usually better to let a common library do this for you. I recommend using https://commons.apache.org/proper/commons-validator/apidocs/org/apache/commons/validator/routines/EmailValidator.html for this. You can find the library here: https://mvnrepository.com/artifact/commons-validator/commons-validator/1.7 and add it to the `pom.xml` file. Implementation code might look something like this:
```
EmailValidator validator = EmailValidator.getInstance();
List<String> validEmails = depositors.stream().filter(e -> validator.isValid(e)).toList();
```

**Review by rushtong — COMMENTED** [review]
Code and tests look good - see suggestion inline.

**Review by rushtong — APPROVED** [review]
Thank you for the changes 👍🏽

File: src/main/java/org/broadinstitute/consent/http/service/VoteService.java [hunk 3]
```diff
@@ -362,22 +366,29 @@ protected void notifyCustodiansOfApprovedDatasets(List<Dataset> datasets, User r
...
-          .filter(p ->
-              (Objects.nonNull(p.getSchemaProperty()) && p.getSchemaProperty()
-                  .equalsIgnoreCase("dataCustodianEmail")) ||
-                  p.getPropertyName().equalsIgnoreCase("Data Depositor"))
-          .map(DatasetProperty::getPropertyValueAsString)
-          .distinct()
-          .toList();
+
+      // Data Custodians
+      List<String> custodianEmails = d.getDataCustodianEmails();
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/consent/http/service/VoteService.java]: Should we also validate the Data Custodian emails?

**Review by fboulnois — COMMENTED** [review]
One question:

File: src/main/java/org/broadinstitute/consent/http/service/VoteService.java (hunk 3 above)
Comment (hams7504) [reply src/main/java/org/broadinstitute/consent/http/service/VoteService.java]: I'm not sure, but it might be good to add it just in case!

**Review by fboulnois — APPROVED** [review]
lgtm 👍

**Pull request merged**